app/controllers/users_controller.rb

```diff
@@ -11,14 +11,35 @@ def new
   
   def create
     @user = User.new(user_params)
+    
     if @user.save
       AppMailer.welcome_new_user(@user).deliver
+      set_relationships if params[:invitation_token].present?
       redirect_to sign_in_path
     else
       render :new
     end    
   end
 
+  def new_with_invitation_token
+    invitation = Invitation.find_by(token: params[:token])
+    if invitation
+      @invitation_token = invitation.token
+      @user = User.new(email: invitation.invitee_email)
+      render :new
+    else
+      redirect_to expired_token_path
+    end    
+  end 
+
+  def set_relationships
```

Comment:
This should be a private method if you are using it on line 17. So only class itself has access to it.